src/tiledb/cloud/vcf/query.py

```diff
@@ -0,0 +1,364 @@
+    config: Optional[Mapping[str, Any]] = None,
+    verbose: bool = False,
+) -> logging.Logger:
+    """
+    Set the default TileDB context, OS environment variables for AWS,
+    and return a logger instance.
+
+    :param config: config dictionary, defaults to None
+    :return: logger instance
+    """
+
+    try:
+        tiledb.default_ctx(config)
+    except tiledb.TileDBError:
+        # Ignore error if the default context was already set
+        pass
+
+    set_aws_context(config)
+
+    level = logging.DEBUG if verbose else logging.INFO
+    logger = get_logger(level)
+
+    logger.debug(
+        "tiledb.cloud=%s, tiledb=%s, libtiledb=%s",
+        tiledb.cloud.version.version,
+        tiledb.version(),
+        tiledb.libtiledb.version(),
+    )
+
+    return logger
+
+
+# --------------------------------------------------------------------
+# UDFs
+# --------------------------------------------------------------------
+
+
+def vcf_query_udf(
+    dataset_uri: str,
+    *,
+    config: Optional[Mapping[str, Any]] = None,
+    attrs: Optional[Union[Sequence[str], str]] = None,
+    regions: Optional[Union[Sequence[str], str]] = None,
+    bed_file: Optional[str] = None,
+    samples: Optional[Union[Sequence[str], str]] = None,
+    region_partition: Optional[Tuple[int, int]] = None,
+    sample_partition: Optional[Tuple[int, int]] = None,
+    memory_budget_mb: int = 1024,
+    af_filter: Optional[str] = None,
+    log_uri: Optional[str] = None,
+    log_id: str = "query",
+    verbose: bool = False,
+) -> pa.table:
+    """
+    Run a query on a TileDB-VCF dataset.
+
+    :param dataset_uri: dataset URI
+    :param config: config dictionary, defaults to None
+    :param attrs: attribute names to read, defaults to None
+    :param regions: genomics regions to read, defaults to None
+    :param bed_file: URI of a BED file containing genomics regions to read, defaults to None
+    :param samples: sample names to read, defaults to None
+    :param region_partition: region partition tuple (0-based indexed, num_partitions), defaults to None
+    :param sample_partition: sample partition tuple (0-based indexed, num_partitions), defaults to None
+    :param memory_budget_mb: VCF memory budget in MiB, defaults to 1024
+    :param af_filter: allele frequency filter, defaults to None
+    :param log_uri: log array URI for profiling, defaults to None
+    :param log_id: profiler event ID, defaults to "query"
+    :param verbose: verbose logging, defaults to False
+    :return: Arrow table containing the query results
+    """
+    import tiledbvcf
+
+    # Adjust inputs
+    if config is None:
+        config = {}
+    if isinstance(attrs, str):
+        attrs = [attrs]
+    if isinstance(regions, str):
+        regions = [regions]
+    if isinstance(samples, str):
+        samples = [samples]
+
+    logger = setup(config, verbose)
+    logger.debug("tiledbvcf=%s", tiledbvcf.version)
+
+    config["rest.use_refactored_array_open"] = True
```

Comment:
When running locally, this will mutate the dictionary that the user passed in (and which might be used by later steps). Is that OK?

Reply:
Good catch. It's probably OK, since local will be used for testing, but it's easy to fix that.